### ISSUE TITLE
Would like to see the running costs (OPEN_AI) for a session with AutoGPT

### ISSUE THREAD

**Issue by timurista** [Significant-Gravitas/Auto-GPT#393]
### Duplicates

- [X] I have searched the existing issues

### Summary 💡

My current challenge is it's hard to know how much it's costing me to run autoGPT for each session (for OPEN_AI in particular)

If I knew how much it was costing me and the current cost per session I could make informed decisions about running in continuous mode or authorizing discrete tasks.

First it would be great just to see estimated costs per session.

Second, I would like to be able to set budget for each session with a soft/hard cap.

For example a soft cap of $2 would mean it leaves continuous mode when it nears $2 estimated cost and you see a warning.
A hard cap would end the session for me, but is optional.
If neither are provided it runs as expected.

Thanks,
Tim

**Comment by koltregaskes** [comment]
Refresh the OpenAI usage price while running?

**Comment by gknerr** [comment]
Yes, a goal could be, don't spend more than 5.00 USD in OpenAI API account for this whatever.

**Comment by Qoyyuum** [comment]
Waiting on PR #762

**Comment by timurista** [comment]
> Waiting on PR #762 

I looked at PR. Thank you, this would be a big help. I do realize the pricing model for open AI  can change in the future so this may not be the last PR in the area of cost estimation. But it's nice to have some idea of the ongoing spend. Appreciate the work!

**Comment by Boostrix** [comment]
In progress as part of: #4040
